packages/cli/src/bin/utils/app-api.ts

```diff
@@ -0,0 +1,200 @@
+import { readFileSync } from 'node:fs';
+import { Spinner } from './spinner.js';
+import { chalk } from './format.js';
+import { AppConfig } from '../../lib/app-config.js';
+
+export type FusionEnv = 'ci' | 'fqa' | 'fprd';
+
+/* build api endpoint url */
+export const getEndpointUrl = (
+    endpoint: string,
+    fusionEnv: FusionEnv,
+    version: string = '1.0-preview',
+) => {
+    const { FUSION_APP_API: appApiUrl } = process.env;
+    return (
+        appApiUrl ??
+        `https://fusion-s-apps-${fusionEnv}.azurewebsites.net/${endpoint}?api-version=${version}`
+    );
+};
+
+export const validateToken = () => {
+    const spinner = Spinner.Current;
+    spinner.info('Validating FUSION_TOKEN');
+
+    if (!process?.env?.FUSION_TOKEN) {
+        spinner.fail('😞', chalk.yellowBright('FUSION_TOKEN not found'));
```

Comment:
be more verbose, imagine debugging a workflow.

Reply:
Done